nexus/db-queries/src/db/datastore/support_bundle.rs

```diff
@@ -0,0 +1,1298 @@
+            .await
+            .expect("Should be able to create bundle");
+
+        db.terminate().await;
+        logctx.cleanup_successful();
+    }
+
+    #[tokio::test]
+    async fn test_crud_operations() {
+        let logctx = dev::test_setup_log("test_crud_operations");
+        let db = TestDatabase::new_with_datastore(&logctx.log).await;
+        let (opctx, datastore) = (db.opctx(), db.datastore());
+
+        let test_sled = create_sled_and_zpools(&datastore, &opctx, 1).await;
+        let reason = "Bundle for test";
+        let this_nexus_id = OmicronZoneUuid::new_v4();
+
+        // Create the bundle, then observe it through the "getter" APIs
+
+        let mut bundle = datastore
+            .support_bundle_create(&opctx, reason, this_nexus_id)
+            .await
+            .expect("Should be able to create bundle");
+        assert_eq!(bundle.reason_for_creation, reason);
+        assert_eq!(bundle.reason_for_failure, None);
+        assert_eq!(bundle.assigned_nexus, Some(this_nexus_id.into()));
+        assert_eq!(bundle.state, SupportBundleState::Collecting);
+        assert_eq!(bundle.zpool_id, test_sled.pools[0].pool.into());
+        assert_eq!(bundle.dataset_id, test_sled.pools[0].dataset.into());
+
+        let observed_bundle = datastore
+            .support_bundle_get(&opctx, bundle.id.into())
+            .await
+            .expect("Should be able to get bundle we just created");
+        // Overwrite this column; it is modified slightly upon database insertion.
+        bundle.time_created = observed_bundle.time_created;
+        assert_eq!(bundle, observed_bundle);
+
+        let pagparams = DataPageParams::max_page();
+        let observed_bundles = datastore
+            .support_bundle_list(&opctx, &pagparams)
+            .await
+            .expect("Should be able to get bundle we just created");
+        assert_eq!(1, observed_bundles.len());
+        assert_eq!(bundle, observed_bundles[0]);
+
+        // Destroy the bundle, observe the new state
+
+        datastore
+            .support_bundle_update(
+                &opctx,
+                bundle.id.into(),
+                SupportBundleState::Destroying,
+            )
+            .await
+            .expect("Should be able to destroy our bundle");
+        let observed_bundle = datastore
+            .support_bundle_get(&opctx, bundle.id.into())
+            .await
+            .expect("Should be able to get bundle we just created");
+        assert_eq!(SupportBundleState::Destroying, observed_bundle.state);
+
+        // Delete the bundle, observe that it's gone
+
+        datastore
+            .support_bundle_delete(&opctx, bundle.id.into())
+            .await
+            .expect("Should be able to destroy our bundle");
+        let observed_bundles = datastore
+            .support_bundle_list(&opctx, &pagparams)
+            .await
+            .expect("Should be able to get bundle we just created");
```

Comment:
Nit: we don't expect to get the bundle we just deleted.

```suggestion
            .expect("Should be able to query when no bundles");
```

Reply:
Updated